Canal/UserControls/FileControl.cs

```diff
@@ -352,6 +353,30 @@ private void CodeBoxOnUndoRedoStateChanged(object sender, EventArgs eventArgs)
             redoButton.Enabled = codeBox.RedoEnabled;
         }
 
+        private void CodeBoxOnSelectionChanged(object sender, EventArgs eventArgs)
+        {
+           tableOfContents.HighlightFirstMatchingNode(FindLastProcedureOrSectionOrDivisonNames(codeBox.Selection.FromLine));
+        }
+
+        /// <summary>
+        /// Filters all possible procedure, section and division names from the given line up to the start of the document.
+        /// This has to return a list of all possible names instead of just the next name as there is no way of finding a 
+        /// regex which distincly filters named names only. This list can be compared to the ToC later and the first match equals the contextual correct name. 
+        /// </summary>
+        /// <param name="lineNumber">The line number to search up from.</param>
+        /// <returns>A list of possible procedure, section and division names from the given line up.</returns>
+        private List<string> FindLastProcedureOrSectionOrDivisonNames(int lineNumber)
+        {
+            List<string> procedureNames = new List<string>();
+            for (int currentLineNumber = lineNumber; currentLineNumber >= 0; currentLineNumber--)
+            {
+                string currenctLineText = codeBox.GetLineText(currentLineNumber);
+                procedureNames.Add(Constants.ProcedureOrSectionOrDivisonRegex.Match(currenctLineText).Groups["name"].Value);

```

Comment:
Nein, da:
`003700 INPUT-OUTPUT SECTION.                                                    
003800 FILE-CONTROL.`
Klickt man in die Zeile mit `FILE-CONTROL` würde der Regex dies als Elements des ToC-Baums erkennen, es ist jedoch nicht vorhanden. Deswegen braucht man an dieser Stelle eine List mit allen möglichen Kandidaten, um sie mit dem Baum abgleichen zu können.